mlkem/indcpa.c

```diff
@@ -155,34 +157,69 @@ static unsigned int rej_uniform(int16_t *r,
 #define GEN_MATRIX_NBLOCKS ((12*KYBER_N/8*(1 << 12)/KYBER_Q + XOF_BLOCKBYTES)/XOF_BLOCKBYTES)
 // Not static for benchmarking
 void gen_matrix(polyvec *a, const uint8_t seed[KYBER_SYMBYTES], int transposed) {
-    unsigned int ctr, i, j, k;
-    unsigned int buflen, off;
-    uint8_t buf[GEN_MATRIX_NBLOCKS * XOF_BLOCKBYTES + 2];
+    unsigned int ctr, i, j;
+    unsigned int buflen;
+    uint8_t bufx[KECCAK_WAY][GEN_MATRIX_NBLOCKS *XOF_BLOCKBYTES + 2];
+    uint8_t *buf = NULL;
+    int16_t *vec = NULL;
     xof_state state;
 
-    for (i = 0; i < KYBER_K; i++) {
-        for (j = 0; j < KYBER_K; j++) {
+    keccakx kx;
+    uint8_t seedxy[KECCAK_WAY][KYBER_SYMBYTES + 2];
+
+    for (i = 0; i < (KYBER_K *KYBER_K / KECCAK_WAY) *KECCAK_WAY; i += KECCAK_WAY) {
+        for (j = 0; j < KECCAK_WAY; j++) {
+            memcpy(seedxy[j], seed, KYBER_SYMBYTES);
             if (transposed) {
-                xof_absorb(&state, seed, i, j);
+                seedxy[j][KYBER_SYMBYTES + 0] = (i + j) / KYBER_K;
+                seedxy[j][KYBER_SYMBYTES + 1] = (i + j) % KYBER_K;
             } else {
-                xof_absorb(&state, seed, j, i);
+                seedxy[j][KYBER_SYMBYTES + 0] = (i + j) % KYBER_K;
+                seedxy[j][KYBER_SYMBYTES + 1] = (i + j) / KYBER_K;
             }
 
-            xof_squeezeblocks(buf, GEN_MATRIX_NBLOCKS, &state);
-            buflen = GEN_MATRIX_NBLOCKS * XOF_BLOCKBYTES;
-            ctr = rej_uniform(a[i].vec[j].coeffs, KYBER_N, buf, buflen);
-
-            while (ctr < KYBER_N) {
-                off = buflen % 3;
-                for (k = 0; k < off; k++) {
-                    buf[k] = buf[buflen - off + k];
```

Comment:
offset related are completely removed. This line can't be reached. See 
https://github.com/pq-crystals/kyber/issues/74

Reply:
Good spot